python/ray/dag/compiled_dag_node.py

```diff
@@ -681,13 +684,10 @@ def __init__(
         # Mapping from the actor handle to the node ID that the actor is on.
         self.actor_to_node_id: Dict["ray.actor.ActorHandle", str] = {}
 
-        # This is set to true when type hint of `transport="nccl"` is used.
-        self._use_default_nccl_group = False
-        # This is set to the specified custom nccl group
-        # if there exists a type hint of `transport=nccl_group`.
-        self._custom_nccl_group_p2p: Optional[GPUCommunicator] = None
-        # The NCCL group ID for P2P send/recv operations.
-        self._nccl_group_id_p2p: Optional[str] = None
+        # The `custom_nccl_group` specified in `experimental_compile`.
+        self._custom_nccl_group_p2p: Optional[GPUCommunicator] = custom_nccl_group
+        # The default NCCL group ID for P2P send/recv operations whose `transport=nccl`.
+        self._default_nccl_group_id_p2p: Optional[str] = None
```

Comment:
Why not just set this equal to the custom group's ID, if one was provided, and None otherwise?

Reply:
Done.